### PR TITLE
Docs: markdown editor known issues

### PR DESCRIPTION
**Pull request by godai78**
Adding known issues to the Markdown output guide and the Autoformatting (input) guide.

Closes: #8520 

May also close: #5721 (this is the final instalment in a series of improvements to Markdown-related guides).

### CODE REVIEW

**Comment by AnnaTomanek** [comment]
Have you checked our feature guides to review other features that haven't been reported by the community yet? How are list styles supported? Image features? What happens with table styles?

**Comment by godai78** [comment]
We did check the community issues and talked it over with QA. However: we should not list issues for features that are simply not covered by Markdown. Things like table or list styles are not supported by the syntax and hence it is not a bug that they are not output - just like the page breaks (that we listed because it was mention as a community issue only). We would have to list all rich-text features then. Otherwise than that only horizontal rule is not supported yet, which I find proper to mention in fact.

**Comment by AnnaTomanek** [comment]
It makes sense. I mean, for us it makes sense that CKE5 cannot output Markdown code for something that Markdown does not support. But for the end users or less aware developers who could think "hey this is a great Markdown editor, it has so many features!!!!" and then configure their editor to have everything with these features available in the toolbar but not really "working" if you check the output, then it would be a bit WTF. So it makes sense to make a note of that and even give some examples.

**Comment by godai78** [comment]
I added a note in the lead and then a large infobox in the Known issues section. I also tested against the horizontal rule which proved it get stripper (this is a possible future implementation candidate). I believe this should be satisfactory now.

File: packages/ckeditor5-markdown-gfm/docs/_snippets/features/markdown.html [hunk 1]
```diff
@@ -3,6 +3,8 @@
 <h2>The Markdown language</h2>
 <p>Markdown is a lightweight markup language that you can use to add formatting elements to plain text documents. Created by <a href="https://daringfireball.net/projects/markdown/" target="_blank">John Gruber</a> in 2004, Markdown is now one of the world&rsquo;s most popular markup languages.</p>
```
Comment (AnnaTomanek) [inline packages/ckeditor5-markdown-gfm/docs/_snippets/features/markdown.html:4]:
```suggestion
<p>Markdown is a lightweight markup language that you can use to add formatting elements to plain text documents. Created by <a href="https://daringfireball.net/projects/markdown/" target="_blank">John Gruber</a> in <s>2003</s>2004, Markdown is now one of the world&rsquo;s most popular markup languages.</p>
```

File: packages/ckeditor5-markdown-gfm/docs/features/markdown.md [hunk 2]
```diff
@@ -5,6 +5,8 @@ category: features
 
 The {@link module:markdown-gfm/markdown~Markdown} plugin allows switching the default CKEditor 5 output from HTML to Markdown. This allows for producing lightweight text documents with a simple formatting syntax, widespread among the programming and development communities and popular in many environments (e.g. GitHub). Coupled with the {@link features/autoformat autoformatting} feature, it allows for the full-fledged Markdown WYSIWYG editing experience.
 
+Please remember that Markdown syntax is very simple and therefore this plugin does not support rich-text output, even if the formatting maybe applied to the content.
```
Comment (AnnaTomanek) [inline packages/ckeditor5-markdown-gfm/docs/features/markdown.md:8]:
```suggestion
Please remember that Markdown syntax is very simple and it does not cover all the rich-text features. Some features provided by CKEditor 5 will thus work as intended only when output to HTML as they have no Markdown equivalent.
```

File: packages/ckeditor5-markdown-gfm/docs/features/markdown.md [hunk 3]
```diff
@@ -81,6 +83,17 @@ ClassicEditor
 
 ```
 
+## Known issues
+* Pasting Markdown-formatted content will not automatically convert the pasted syntax markers into properly formatted content currently. See these GitHub issues for possible future development: [#2321](https://github.com/ckeditor/ckeditor5/issues/2321), [#2322](https://github.com/ckeditor/ckeditor5/issues/2322).
```
Comment (AnnaTomanek) [inline packages/ckeditor5-markdown-gfm/docs/features/markdown.md:87]:
```suggestion
While the Markdown plugin is stable and ready to use, some issues were reported for it. Feel free to upvote 👍 them on GitHub if they are important for you:

* Pasting Markdown-formatted content does not automatically convert the pasted syntax markers into properly formatted content. See these GitHub issues for possible future development: [#2321](https://github.com/ckeditor/ckeditor5/issues/2321), [#2322](https://github.com/ckeditor/ckeditor5/issues/2322).
```

File: packages/ckeditor5-markdown-gfm/docs/features/markdown.md [hunk 4]
```diff
@@ -81,6 +83,17 @@ ClassicEditor
...
+* The horizontal rule is not supported by the {@link features/autoformat autoformatting} but it does get output if inserted via WYSIWYG interface.
```
Comment (AnnaTomanek) [inline packages/ckeditor5-markdown-gfm/docs/features/markdown.md:88]:
```suggestion
* The horizontal rule is not supported by the {@link features/autoformat autoformatting} feature but it gets output to Markdown if inserted using the toolbar button.
```

File: packages/ckeditor5-markdown-gfm/docs/features/markdown.md [hunk 5]
```diff
@@ -81,6 +83,17 @@ ClassicEditor
...
+
+<info-box info>
+	Please bear in mind that the Markdown output processor will not produce any rich-text content. The [Markdown syntax](https://daringfireball.net/projects/markdown/syntax) is very simple and only supports limited formatting.
```
Comment (AnnaTomanek) [inline packages/ckeditor5-markdown-gfm/docs/features/markdown.md:91]:
```suggestion
	Please bear in mind that the Markdown data processor does not support all rich text features. The [Markdown syntax](https://daringfireball.net/projects/markdown/syntax) is very simple and only supports limited formatting options.
```

File: packages/ckeditor5-markdown-gfm/docs/features/markdown.md [hunk 6]
```diff
@@ -81,6 +83,17 @@ ClassicEditor
...
+
+	This means than advanced formatting, like list styles, tables or page break markers will be stripped in the effecting data. These are not supported by Markdown and therefore cannot be reproduced. This is not a bug.
```
Comment (AnnaTomanek) [inline packages/ckeditor5-markdown-gfm/docs/features/markdown.md:93]:
```suggestion
	This means that advanced formatting like list styles, table styles or page break markers will be stripped in the effecting data. These are not supported by Markdown and therefore cannot be converted from HTML to Markdown.
```

**Review by AnnaTomanek — COMMENTED** [review]
Just for the sake of checking it out, have you tried how tables work when output to Markdown? Creating tables in Markdown is a PITA, so maybe it would be nice to have them in the sample.

**Comment by godai78** [comment]
> Just for the sake of checking it out, have you tried how tables work when output to Markdown? Creating tables in Markdown is a PITA, so maybe it would be nice to have them in the sample.

They do work and they are... well, let me show. They get stripped. I had this improperly expressed in the docs as I see now. I'm totally not for having this in the sample, but I can do it in like 2 minutes.

**Comment by godai78** [comment]
But sometimes the tables also get output as HTML. @ckeditor/qa-team can you take a look?

**Comment by Mgsy** [comment]
> But sometimes the tables also get output as HTML. [**@ckeditor/qa-team**](https://github.com/orgs/ckeditor/teams/qa-team) can you take a look?

Nice catch. It looks like plugin returns a proper table in markdown if it contains a table header (`<thead>` element). A basic table is not recognized and returned as HTML. 

I've created an issue - #8572.

File: packages/ckeditor5-markdown-gfm/docs/_snippets/features/markdown.html [hunk 7]
```diff
@@ -32,6 +34,36 @@ <h2>Formatting blocks with Markdown</h2>
 	</li>
 </ul>
 
+<h2>Tables are supported, too</h2>
```
Comment (AnnaTomanek) [inline packages/ckeditor5-markdown-gfm/docs/_snippets/features/markdown.html:37]:
```suggestion
<h2>Support for tables in Markdown</h2>
```

File: packages/ckeditor5-markdown-gfm/docs/_snippets/features/markdown.html [hunk 8]
```diff
@@ -32,6 +34,36 @@ <h2>Formatting blocks with Markdown</h2>
...
+<p>But bear in mind, they get stripped of the formatting</p>
```
Comment (AnnaTomanek) [inline packages/ckeditor5-markdown-gfm/docs/_snippets/features/markdown.html:38]:
```suggestion
<p>Bear in mind that Markdown has only very basic support for tables, so things like table styles or merged cells will not work.</p>
```

File: packages/ckeditor5-markdown-gfm/docs/features/markdown.md [hunk 9]
```diff
@@ -81,6 +83,20 @@ ClassicEditor
...
+
+<info-box info>
+	Please bear in mind that the Markdown data processor does not support all rich text features. The [Markdown syntax](https://daringfireball.net/projects/markdown/syntax) is very simple and only supports limited formatting options.
+
+	This means that advanced formatting like list styles, table styles or page break markers will be stripped in the effecting data. These are not supported by Markdown and therefore cannot be converted from HTML to Markdown.
+</info-box>
+
+While the Markdown plugin is stable and ready to use, some issues were reported for it. Feel free to upvote 👍&nbsp; them on GitHub if they are important for you:
+
+* The horizontal rule is not yet supported by the {@link features/autoformat autoformatting} feature but it gets output to Markdown nevertheless if inserted using the toolbar button.
```
Comment (AnnaTomanek) [inline packages/ckeditor5-markdown-gfm/docs/features/markdown.md:96]: Issue number?

File: packages/ckeditor5-markdown-gfm/docs/features/markdown.md [hunk 10]
```diff
@@ -81,6 +83,20 @@ ClassicEditor
...
+* Pasting Markdown-formatted content does not automatically convert the pasted syntax markers into properly formatted content. See these GitHub issues for possible future development: [#2321](https://github.com/ckeditor/ckeditor5/issues/2321), [#2322](https://github.com/ckeditor/ckeditor5/issues/2322).
+* Tables would render as HTML output if there is no table head defined. See the current status of this issue on GitHub: [#8575](https://github.com/ckeditor/ckeditor5/issues/8572).
```
Comment (AnnaTomanek) [inline packages/ckeditor5-markdown-gfm/docs/features/markdown.md:98]:
```suggestion
* Tables will render as HTML output if there is no table head defined. GitHub issue: [#8575](https://github.com/ckeditor/ckeditor5/issues/8572).
```

File: packages/ckeditor5-markdown-gfm/docs/features/markdown.md [hunk 11]
```diff
@@ -81,6 +83,20 @@ ClassicEditor
...
+<info-box info>
+	Please bear in mind that the Markdown data processor does not support all rich text features. The [Markdown syntax](https://daringfireball.net/projects/markdown/syntax) is very simple and only supports limited formatting options.
+
+	This means that advanced formatting like list styles, table styles or page break markers will be stripped in the effecting data. These are not supported by Markdown and therefore cannot be converted from HTML to Markdown.
+</info-box>
+
+While the Markdown plugin is stable and ready to use, some issues were reported for it. Feel free to upvote 👍&nbsp; them on GitHub if they are important for you:
+
+* The horizontal rule is not yet supported by the {@link features/autoformat autoformatting} feature but it gets output to Markdown nevertheless if inserted using the toolbar button.
+* Pasting Markdown-formatted content does not automatically convert the pasted syntax markers into properly formatted content. See these GitHub issues for possible future development: [#2321](https://github.com/ckeditor/ckeditor5/issues/2321), [#2322](https://github.com/ckeditor/ckeditor5/issues/2322).
```
Comment (AnnaTomanek) [inline packages/ckeditor5-markdown-gfm/docs/features/markdown.md:97]:
```suggestion
* Pasting Markdown-formatted content does not automatically convert the pasted syntax markers into properly formatted content. GitHub issues: [#2321](https://github.com/ckeditor/ckeditor5/issues/2321), [#2322](https://github.com/ckeditor/ckeditor5/issues/2322).
```

File: packages/ckeditor5-autoformat/docs/features/autoformat.md [hunk 12]
```diff
@@ -89,6 +89,11 @@ The {@link module:autoformat/autoformat~Autoformat} feature bases on {@link modu
 
 You can use these tools to create your own autoformatters. Check the [`Autoformat` feature's code](https://github.com/ckeditor/ckeditor5/blob/master/packages/ckeditor5-autoformat/src/autoformat.js) as an example.
 
+## Known issues
+* Using underscore inline (eg. variable names like this_variable) may result in italics. GitHub issue: [#2388](https://github.com/ckeditor/ckeditor5/issues/2388).
```
Comment (AnnaTomanek) [inline packages/ckeditor5-autoformat/docs/features/autoformat.md:93]:
```suggestion
* Using underscore inline (eg. variable names like `this_variable`) may result in italics. GitHub issue: [#2388](https://github.com/ckeditor/ckeditor5/issues/2388).
```

File: packages/ckeditor5-autoformat/docs/features/autoformat.md [hunk 13]
```diff
@@ -89,6 +89,11 @@ The {@link module:autoformat/autoformat~Autoformat} feature bases on {@link modu
...
+* Pasting Markdown-formatted content will not automatically convert the pasted syntax markers into properly formatted content currently. See these GitHub issues for possible future development: [#2321](https://github.com/ckeditor/ckeditor5/issues/2321), [#2322](https://github.com/ckeditor/ckeditor5/issues/2322).
```
Comment (AnnaTomanek) [inline packages/ckeditor5-autoformat/docs/features/autoformat.md:94]:
```suggestion
* Pasting Markdown-formatted content does not automatically convert the pasted syntax markers into properly formatted content. GitHub issues: [#2321](https://github.com/ckeditor/ckeditor5/issues/2321), [#2322](https://github.com/ckeditor/ckeditor5/issues/2322).
```

File: packages/ckeditor5-autoformat/docs/features/autoformat.md [hunk 14]
```diff
@@ -89,6 +89,11 @@ The {@link module:autoformat/autoformat~Autoformat} feature bases on {@link modu
...
+* The feature does not support adding horizontal rules currently. See this GitHub issue for possible future development: [#5720](https://github.com/ckeditor/ckeditor5/issues/5720).
```
Comment (AnnaTomanek) [inline packages/ckeditor5-autoformat/docs/features/autoformat.md:95]:
```suggestion
* At the moment the feature does not support adding horizontal rules. GitHub issue: [#5720](https://github.com/ckeditor/ckeditor5/issues/5720).
```

File: packages/ckeditor5-autoformat/docs/features/autoformat.md [hunk 15]
```diff
@@ -89,6 +89,11 @@ The {@link module:autoformat/autoformat~Autoformat} feature bases on {@link modu
 
 You can use these tools to create your own autoformatters. Check the [`Autoformat` feature's code](https://github.com/ckeditor/ckeditor5/blob/master/packages/ckeditor5-autoformat/src/autoformat.js) as an example.
 
+## Known issues
```
Comment (AnnaTomanek) [inline packages/ckeditor5-autoformat/docs/features/autoformat.md:92]:
```suggestion
## Known issues

While the autoformatting feature is stable and ready to use, some issues were reported for it. Feel free to upvote 👍&nbsp; them on GitHub if they are important for you:
```

**Review by AnnaTomanek — APPROVED** [review]
Some minor and hopefully final suggestions. And let's merge it yesterday.

File: packages/ckeditor5-markdown-gfm/docs/features/markdown.md (hunk 9 above)
Comment (godai78) [reply packages/ckeditor5-markdown-gfm/docs/features/markdown.md:96]:
```suggestion
* The horizontal rule is not yet supported by the {@link features/autoformat autoformatting} feature but it gets output to Markdown nevertheless if inserted using the toolbar button. GitHub issue: [#5720](https://github.com/ckeditor/ckeditor5/issues/5720).
```

Comment (godai78) [reply packages/ckeditor5-markdown-gfm/docs/features/markdown.md]: Added.

**Pull request merged**